helpers/student.js

```diff
@@ -95,13 +96,76 @@ exports.image = async function (req, res, next) {
     else {
       updatedFields = { profilePicURL: imageURL }
     }
-    const updatedStudent = await findAndUpdate(decodedToken.email, updatedFields);
-    res.send({ updatedStudent });
+    const updatedStudent = await studentModel.findOneAndUpdate({ _id: decoded.id }, updatedFields);
+    return res.status(200).json(updatedStudent)
   }
   catch (err) {
-    return res.json({ message: err.message });
+    return res.status(400).json({ message: err.message });
+  }
+}
+
+exports.getClubs = async function (req, res) {
+
+  // pull id from jwt
+  const token = req.headers.authorization.split(" ")[1];
+  const decoded = jwt.decode(token, { complete: true });
+  let id = decoded.payload.id;
+  console.log('Request made from:', id)
+
+  // Find which clubs the student follows
+  let clubs
+  try {
+    clubs = await studentModel.findOne({ _id: id }, 'clubs');
+  }
+  catch (err) {
+    return res.status(400).json({
+      message: err.message
+    })
+  }
+
+  let clubDetails = []
+
+  for (const club of clubs.clubs) {
+    const clubQuery = await clubModel.findOne({ email: club })
```

Comment:
I was unable to extract id from club, and didn't know how to change the async for loop. I figure this is a change we would like to make in the long run though.